src/main/java/com/homework/lesson14/flyweight/Factory.java

```diff
@@ -0,0 +1,20 @@
+package com.homework.lesson14.flyweight;
+
+public class Factory {
+    private Flyweight[] pool;
+
+    public Factory(int maxRows) {
+        if (maxRows >= 0) {
+            pool = new Flyweight[maxRows];
```

Comment:
In case maxRows<0 better to initialize by some default value or even throw a custom exception.